packages/extension-commerce-bm-seo/src/components/seo-hoc.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useState, useEffect, useRef} from 'react'
+import {useBlockNavigation, useRoutes} from '@salesforce/pwa-kit-react-sdk/ssr/universal/hooks'
+import {useUrlMapping} from '@salesforce/commerce-sdk-react'
+import {useLocation, Redirect} from 'react-router-dom'
+import {useApplicationExtensionsStore} from '@salesforce/pwa-kit-extension-sdk/react'
+import {useExtensionConfig} from '../hooks/use-extension-config'
+type SeoHOCProps = React.ComponentPropsWithoutRef<any>
+
+interface UrlMappingResponse {
+    resourceType?: string
+    resourceId?: string
+    destinationUrl?: string
+}
+
+const getComponent = (
+    routes: Array<{component: React.ComponentType<any> | undefined; path: string}>,
+    resourceTypeToComponentMap: {[key: string]: string},
+    resourceType: string
+) => {
+    const ComponentClass = routes.find((_route) =>
+        _route.component?.displayName?.includes(resourceTypeToComponentMap[resourceType])
+    )?.component
+    return ComponentClass
+}
```

Comment:
Lets move this to a utils file or sorts. Also add some comments at to what this function does. We are trying to have better file organization and trying to keep 1 method/function per file when we can.